pkg/healthcheck/healthcheck.go

```diff
@@ -2098,12 +2092,19 @@ func (hc *HealthChecker) checkValidatingWebhookConfigurations(ctx context.Contex
 	return checkResources("ValidatingWebhookConfigurations", objects, []string{k8s.SPValidatorWebhookConfigName}, shouldExist)
 }
 
-func (hc *HealthChecker) checkProxyInitRunsAsRoot(ctx context.Context, config *l5dcharts.Values) error {
-	runAsRoot := config != nil && config.ProxyInit != nil && config.ProxyInit.RunAsRoot
+// CheckProxyInitRunsAsRoot checks that proxyInit will run as root if cluster
+// uses the Docker container runtime.
+func CheckProxyInitRunsAsRoot(ctx context.Context, k8sAPI *k8s.KubernetesAPI, runAsRoot bool) error {
+	// If the cluster is ignored (and the k8s API is not initialized) or the
+	// proxyInit container will run as root, there is no validation that needs
+	// to take place.
+	if k8sAPI == nil || runAsRoot {
```

Comment:
Ah, I see that we just return a success if the k8sAPI is nil.  I'd recommend pushing this logic up so that this function always expects that the k8sAPI is non-nil and the calling scope should only call this function if it actually needs to do the check.